misc/python/materialize/mzbuild.py

```diff
@@ -514,24 +505,33 @@ def build(self) -> None:
         ]
         spawn.runv(cmd, stdin=f, stdout=sys.stderr.buffer)
 
-    def acquire(self) -> AcquiredFrom:
-        """Download or build the image.
-
-        Returns:
-            acquired_from: How the image was acquired.
-        """
+    def acquire(self) -> None:
+        """Download or build the image if it does not exist locally."""
+        if self.acquired:
+            return
 
+        ui.header(f"Acquiring {self.spec()}")
         try:
             spawn.runv(
                 ["docker", "pull", self.spec()],
                 stdout=sys.stderr.buffer,
             )
-            return AcquiredFrom.REGISTRY
         except subprocess.CalledProcessError:
-            pass
+            for dep in self.dependencies.values():
+                dep.acquire()
```

Comment:
If this loop moves in build (see next comment) we can just call `self.build()` here